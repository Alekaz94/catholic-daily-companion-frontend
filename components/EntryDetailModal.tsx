import React from 'react';
import { Modal, View, Text, TouchableOpacity } from "react-native";
import { JournalEntry } from "../models/JournalEntry";
import { Typography } from '../styles/Typography';
import { Layout } from '../styles/Layout';
import { AppTheme } from '../styles/colors';

interface Props {
    visible: boolean;
    entry: JournalEntry | null;
    onClose: () => void;
}

const EntryDetailModal: React.FC<Props> = ({visible, entry, onClose}) => {
    if(!entry) {
        return null;
    }

    return (
        <Modal visible={visible} animationType='slide'>
            <View style={[Layout.container, {backgroundColor: AppTheme.journal.background}]}>
                <Text style={[Typography.title, {color: AppTheme.journal.text}]}>{entry.title}</Text>
                <Text style={[Typography.small, {color: AppTheme.journal.text}]}>{entry.date}</Text>
                <Text style={[Typography.body, {marginTop: 20, marginBottom: 20, color: AppTheme.journal.text}]}>{entry.content}</Text>

                <TouchableOpacity onPress={onClose} style={[Layout.button, {width: "50%", alignSelf: "center", backgroundColor: "#B794F4"}]}>
                    <Text style={[Layout.buttonText, {alignSelf: "center", color: AppTheme.journal.text}]}>Close</Text>
                </TouchableOpacity> 
            </View>
        </Modal>
    );
}

export default EntryDetailModal;