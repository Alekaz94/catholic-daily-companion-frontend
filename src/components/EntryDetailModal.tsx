import React from 'react';
import { Modal, View, Text, TouchableOpacity } from "react-native";
import { JournalEntry } from "../models/JournalEntry";
import { Typography } from '../styles/Typography';
import { Layout } from '../styles/Layout';
import Divider from './Divider';
import { useAppTheme } from '../hooks/useAppTheme';

interface Props {
    visible: boolean;
    entry: JournalEntry | null;
    onClose: () => void;
}

const EntryDetailModal: React.FC<Props> = ({visible, entry, onClose}) => {
    const theme = useAppTheme();

    if(!entry) {
        return null;
    }

    return (
        <Modal visible={visible} animationType='slide'>
            <View style={[Layout.container, {backgroundColor: theme.journal.background}]}>
                <Text style={[Typography.title, {color: theme.journal.text, fontWeight: "bold", textAlign: "center"}]}>{entry.title}</Text>
                <Divider />
                <Text style={[Typography.label, {color: theme.journal.text}]}>Created: {entry.date}</Text>
                {entry.updatedAt !== entry.date && 
                    <Text style={[Typography.label, {color: theme.journal.text}]}>Updated: {entry.updatedAt}</Text>
                }
                <Divider />
                <Text style={[Typography.body, {marginTop: 20, marginBottom: 20, color: theme.journal.text, fontSize: 18}]}>{entry.content}</Text>

                <TouchableOpacity onPress={onClose} style={[Layout.button, {width: "50%", alignSelf: "center", backgroundColor: theme.journal.cardOne, borderWidth: 1}]}>
                    <Text style={[Layout.buttonText, {alignSelf: "center", color: theme.journal.text}]}>Close</Text>
                </TouchableOpacity> 
            </View>
        </Modal>
    );
}

export default EntryDetailModal;