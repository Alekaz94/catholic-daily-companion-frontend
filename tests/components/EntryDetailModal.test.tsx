import { act, fireEvent, render } from "@testing-library/react-native";
import EntryDetailModal from "../../components/EntryDetailModal"

const mockJournalEntry = {
    id: "1sad23",
    date: "2025-08-25",
    updatedAt: "2025-08-30",
    title: "Test entry",
    content: "This is the content of the test entry"
}

describe("EntryDetailModal", () => {
    it("does not render if entry is null", () => {
        const { toJSON } = render(
            <EntryDetailModal visible={true} entry={null} onClose={jest.fn()} />
        );
        expect(toJSON()).toBeNull();
    })

    it("does not render content if visible is false", () => {
        const { queryByText } = render(
            <EntryDetailModal visible={false} entry={mockJournalEntry} onClose={jest.fn()} />
        );
    
        expect(queryByText("Test entry")).toBeNull();
    });

    it("renders entry data correctly", () => {
        const { getByText } = render(
            <EntryDetailModal visible={true} entry={mockJournalEntry} onClose={jest.fn()} />
        )
        expect(getByText("Test entry")).toBeTruthy();
        expect(getByText("2025-08-30")).toBeTruthy();
        expect(getByText("This is the content of the test entry")).toBeTruthy();
    })

    it("calls onClose when Close button is pressed", async () => {
        const onCloseMock = jest.fn();
        const { getByText} = render(
            <EntryDetailModal visible={true} entry={mockJournalEntry} onClose={onCloseMock} />
        );

        await act(async () => {
            fireEvent.press(getByText("Close"));
        });
        expect(onCloseMock).toHaveBeenCalled();
    });

    it("can handle multiple close button presses", async () => {
        const onCloseMock = jest.fn();
        const { getByText } = render(
            <EntryDetailModal visible={true} entry={mockJournalEntry} onClose={onCloseMock} />
        );
    
        const closeButton = getByText("Close");
    
        await act(async () => {
            fireEvent.press(closeButton);
            fireEvent.press(closeButton);
        });
    
        expect(onCloseMock).toHaveBeenCalledTimes(2);
    });

    it("matches the snapshot", () => {
        const { toJSON } = render(
            <EntryDetailModal visible={true} entry={mockJournalEntry} onClose={jest.fn()} />
        );
        expect(toJSON()).toMatchSnapshot();
    });
})