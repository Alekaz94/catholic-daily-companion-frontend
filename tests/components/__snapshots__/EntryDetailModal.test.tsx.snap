// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryDetailModal matches the snapshot 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  visible={true}
>
  <View
    style={
      [
        {
          "flex": 1,
          "padding": 20,
        },
        {
          "backgroundColor": "#EDE9FE",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#222",
            "fontSize": 24,
            "fontWeight": "bold",
            "marginBottom": 16,
          },
          {
            "color": "#1A1A1A",
          },
        ]
      }
    >
      Test entry
    </Text>
    <Text
      style={
        [
          {
            "color": "white",
            "fontSize": 12,
          },
          {
            "color": "#1A1A1A",
          },
        ]
      }
    >
      2025-08-30
    </Text>
    <Text
      style={
        [
          {
            "color": "white",
            "fontSize": 14,
          },
          {
            "color": "#1A1A1A",
            "marginBottom": 20,
            "marginTop": 20,
          },
        ]
      }
    >
      This is the content of the test entry
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#B794F4",
          "borderRadius": 8,
          "marginTop": 10,
          "opacity": 1,
          "paddingVertical": 12,
          "width": "50%",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "bold",
            },
            {
              "alignSelf": "center",
              "color": "#1A1A1A",
            },
          ]
        }
      >
        Close
      </Text>
    </View>
  </View>
</Modal>
`;
